### PR TITLE
update docs example with shared eval step

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Popped up in Slack, mentioned by Sidhant Sundrani. 

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Review by kaushikb11 — APPROVED** [review]

File: docs/source/common/lightning_module.rst [hunk 1]
```diff
@@ -586,33 +586,37 @@ For cases like production, you might want to iterate different models inside a L
...
-            metrics = {'test_acc': metrics['val_acc'], 'test_loss': metrics['val_loss']}
+        def test_step(self, batch, batch_idx):
+            loss, acc = self.shared_eval_step(batch, batch_idx)
+            metrics = {'test_acc': acc, 'test_loss': loss}
             self.log_dict(metrics)
+            return metrics
 
-         def configure_optimizers(self):
-             return torch.optim.Adam(self.model.parameters(), lr=0.02)
+        def shared_eval_step(self, batch, batch_idx):
```
Comment (ananthsub) [inline docs/source/common/lightning_module.rst:611]:
```suggestion
        def _shared_eval_step(self, batch, batch_idx):
```
to indicate its a helper and not required from the API?

**Review by ananthsub — APPROVED** [review]

File: docs/source/common/lightning_module.rst [hunk 2]
```diff
@@ -586,33 +586,37 @@ For cases like production, you might want to iterate different models inside a L
...
             return metrics
 
-         def test_step(self, batch, batch_idx):
-            metrics = self.validation_step(batch, batch_idx)
-            metrics = {'test_acc': metrics['val_acc'], 'test_loss': metrics['val_loss']}
+        def test_step(self, batch, batch_idx):
+            loss, acc = self.shared_eval_step(batch, batch_idx)
+            metrics = {'test_acc': acc, 'test_loss': loss}
             self.log_dict(metrics)
+            return metrics
```
Comment (Borda) [inline docs/source/common/lightning_module.rst:609]: why the return here?

**Review by justusschock — APPROVED** [review]

File: docs/source/common/lightning_module.rst (hunk 2 above)
Comment (awaelchli) [reply docs/source/common/lightning_module.rst:609]: it's optional

File: docs/source/common/lightning_module.rst [hunk 3]
```diff
@@ -586,33 +586,37 @@ For cases like production, you might want to iterate different models inside a L
...
+            loss, acc = self.shared_eval_step(batch, batch_idx)
             metrics = {'val_acc': acc, 'val_loss': loss}
             self.log_dict(metrics)
             return metrics
 
-         def test_step(self, batch, batch_idx):
-            metrics = self.validation_step(batch, batch_idx)
-            metrics = {'test_acc': metrics['val_acc'], 'test_loss': metrics['val_loss']}
+        def test_step(self, batch, batch_idx):
+            loss, acc = self.shared_eval_step(batch, batch_idx)
```
Comment (awaelchli) [inline docs/source/common/lightning_module.rst:606]:
```suggestion
            loss, acc = self._shared_eval_step(batch, batch_idx)
```

File: docs/source/common/lightning_module.rst [hunk 4]
```diff
@@ -586,33 +586,37 @@ For cases like production, you might want to iterate different models inside a L
...
-         def validation_step(self, batch, batch_idx):
+        def training_step(self, batch, batch_idx):
             x, y = batch
             y_hat = self.model(x)
             loss = F.cross_entropy(y_hat, y)
-            acc = FM.accuracy(y_hat, y)
+            return loss
 
+        def validation_step(self, batch, batch_idx):
+            loss, acc = self.shared_eval_step(batch, batch_idx)
```
Comment (awaelchli) [inline docs/source/common/lightning_module.rst:600]:
```suggestion
            loss, acc = self._shared_eval_step(batch, batch_idx)
```

**Review by SkafteNicki — APPROVED** [review]

**Review by s-rog — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !
